src/encoder.rs

```diff
@@ -2474,7 +2477,14 @@ fn encode_partition_bottomup(
             cw.bc.update_partition_context(bo, subsize, bsize);
         }
     }
-    (best_rd, Some(best_decision))
+
+    rdo_output.rd_cost = best_rd;
+    rdo_output.part_type = best_partition;
+
+    if best_partition != PartitionType::PARTITION_NONE {
+        rdo_output.part_modes = Vec::new();
```

Comment:
```suggestion
        rdo_output.part_modes.clear();
```
would spare an allocation.

Reply:
Indeed, that is the right way to go. Thanks!